sql/core/src/test/scala/org/apache/spark/sql/CollationSQLRegexpSuite.scala

```diff
@@ -96,6 +98,36 @@ class CollationSQLRegexpSuite
     }
   }
 
+  test("RegExpReplace throws the right exception when replace fails on a particular row") {
```

Comment:
Why did you place this test to collation related test suite, I wonder.

Reply:
Good question. I didn't want to create a whole new test suite for this small test. Therefore, it put it in the only SQL test suite I could find for regexp-expressions. The other `RegexpExpressionsSuite.scala` doesn't do end-to-end SQL testing.

Reply:
How about https://github.com/apache/spark/blob/a1fc7e66883ac61abfd4be15f50455d03cac040c/sql/core/src/test/scala/org/apache/spark/sql/StringFunctionsSuite.scala#L180

Reply:
Thanks, done!